src/Hooks.ts

```diff
@@ -143,3 +145,75 @@ export function useFocusState() {
 
   return focusState;
 }
+
+type EffectCallback = (() => void) | (() => () => void);
+
+// Inspired by same hook from react-navigation v5
+// See https://github.com/react-navigation/hooks/issues/39#issuecomment-534694135
+export const useFocusEffect = (callback: EffectCallback) => {
+  const navigation = useNavigation();
+  const getCallback = useGetter(callback);
+
+  useEffect(() => {
+    let isFocused = false;
+    let cleanup: (() => void) | void;
+
+    if (navigation.isFocused()) {
+      cleanup = getCallback()();
+      isFocused = true;
+    }
+
+    const focusSubscription = navigation.addListener('willFocus', () => {
+      // If callback was already called for focus, avoid calling it again
+      // The focus event may also fire on intial render, so we guard against runing the effect twice
+      if (isFocused) {
+        return;
+      }
+
+      cleanup && cleanup();
+      cleanup = getCallback()();
+      isFocused = true;
+    });
+
+    const blurSubscription = navigation.addListener('willBlur', () => {
+      cleanup && cleanup();
+      cleanup = undefined;
+      isFocused = false;
+    });
+
+    return () => {
+      cleanup && cleanup();
+      focusSubscription.remove();
+      blurSubscription.remove();
+    };
+  }, [getCallback, navigation]);
+};
+
+export const useIsFocused = () => {
+  const navigation = useNavigation();
+  const [focused, setFocused] = useState(navigation.isFocused());
+
+  useEffect(() => {
+    const focusSubscription = navigation.addListener('willFocus', () =>
+      setFocused(true),
+    );
+    const blurSubscription = navigation.addListener('willBlur', () =>
+      setFocused(false),
+    );
+    return () => {
+      focusSubscription.remove();
+      blurSubscription.remove();
+    };
+  }, [setFocused]);
```

Comment:
I think the dep array needs to include `navigation`:

```suggestion
  }, [navigation]);
```

AFAIK React ensures that `setFocused` will have same reference between re-renders (and the ESLint rule doesn't add it either).

Reply:
ah yeah I could include navigation and remove setFocused, didn't know the eslint rule allowed to not provide setFocused

but navigation changes often, so will make a stable getter with useGetter() which will prevent effect from re-rerunning and unsub/resub on each nav (even if it's probably fast anyway)